packages/react-dom-bindings/src/client/ReactDOMComponent.js

```diff
@@ -2485,51 +2485,41 @@ function diffHydratedGenericElement(
       case 'src':
       case 'href':
         if (
+          __DEV__ &&
           value === '' &&
           // <a href=""> is fine for "reload" links.
           !(tag === 'a' && propKey === 'href') &&
           !(tag === 'object' && propKey === 'data')
```

Comment:
how could line could be hit? am i losing or is `propKey` always `href` here?

Reply:
seems silly that we're checking `propKey === 'href'` here too, maybe i'm missing something or maybe that's why the other line was missed?

Reply:
There's a fallthrough case above. The propKey can be either `'data'`, `'src'`' or `'href'`.

Reply:
o rite ok